### DISCUSSION TITLE
Changing dynamically GridCellLayer grids to be drawn is specific projetion

### DISCUSSION THREAD

**Discussion by LiderisPeteris** [visgl/deck.gl#6897] [Q&A] [answered]
Hello!

We are currently exploring ways on how to use [deck.gl](http://deck.gl/) in our company. Currently facing problem with GridCellLayer.

How can we achieve effect, so GridCellLayer grids that are drawn are drawn is specific projetion?

As in example: current.png draws grids as is. What we need is to get desired.png effect so grids are transformed to EPSG:3035 (or any other projection).

Is it possible to somehow transform geometry based on projection?

current.png:
![current](https://user-images.githubusercontent.com/105269750/167573674-c1e82e77-325c-4975-92ec-e8d623b084e0.png)

desired.png:
![desired](https://user-images.githubusercontent.com/105269750/167573703-4582b446-f82c-4af5-9c40-e4e4d1c3e70f.png)

Example of how we would like to change projection to grids: https://viglino.github.io/ol-ext/examples/layer/map.layer.gridbin.html

**Comment by Pessimistress** [comment, marked as answer]
As of v8 deck.gl doesn't support geospatial projections other than Web Mercator. You have two options to implement this:

- Perform the projection on CPU and use `COORDINATE_SYSTEM.CARTESIAN` to render your geometries, see https://codepen.io/Pessimistress/pen/ExYdjeq
- Write a custom shader that projects lon/lat coordinates to another projection system on the GPU.

Either way you will need to create a [custom layer](https://deck.gl/docs/developer-guide/custom-layers). You can look into subclassing the GridCell Layer as a start.

**Reply by LiderisPeteris** [reply]
Thank you very much.
I guess custom shader is the proper way to achieve our goal here.
